code/modules/submaps/submap_archetype.dm

```diff
@@ -13,12 +13,22 @@
 
 /decl/submap_archetype/validate()
 	. = ..()
-	if(!descriptor)
-		. += "no descriptor set"
+	if(!name)
+		. += "no name set"
+	else
+		var/static/list/submaps_by_name
```

Comment:
i don't think this is a good idea, it'll loop over N submaps in N submaps' `validate()` proc. this was a uniqueness unit test for a reason